jhhalchemy/model/time_order.py

```diff
@@ -52,3 +52,48 @@ def read_time_range(cls, *args, **kwargs):
         if end is not None:
             criteria.append(cls.time_order >= -end)
         return cls.read(*criteria)
+
+
+"""
+Helper Function for APIs that interact with time_order models
+"""
+
+
+class InvalidTimestampRange(Exception):
+    """
+    Raise this when start_timestamp > end_timestamp
+    """
+    pass
+
+
+def get_by_range(model_cls, *args, **kwargs):
+    """
+    Get ordered list of models for the specified time range.
+    The timestamp on the earliest model will likely occur before start_timestamp. This is to ensure that we return
+    the models for the entire range.
+
+    :param model_cls: the class of the model to return
+    :param args: arguments specific to the model class
+    :param kwargs: start_timestamp and end_timestamp (see below) as well as keyword args specific to the model class
+    :keyword start_timestamp: the most recent models set before this and all after, defaults to 0
+    :keyword end_timestamp: only models set before (and including) this timestamp, defaults to now
+    :return: model generator
+    """
+    start_timestamp = kwargs.get('start_timestamp')
+    end_timestamp = kwargs.get('end_timestamp')
+    if (start_timestamp is not None) and (end_timestamp is not None) and (start_timestamp > end_timestamp):
```

Comment:
Does `>=` also constitute an InvalidTimestampRange?

Reply:
No because the endpoints are inclusive, so start == end would be the row with that exact time.